### PR TITLE
cpdp-471: handled already subscribed response from subscription create

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/StartController.scala [hunk 1]
```diff
@@ -69,6 +69,9 @@ class StartController @Inject()(
       case (_, Some(_: Subscribed)) =>
         Redirect(routes.HomeController.homepage())
 
+      case (_, Some(AlreadySubsribedWithDifferentGGAccount)) =>
```
Comment (co-category) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/StartController.scala]: typo in  `AlreadySubsribedWithDifferentGGAccount`

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/StartController.scala]: fixed

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/SubscriptionController.scala [hunk 2]
```diff
@@ -63,33 +64,45 @@ class SubscriptionController @Inject()(
...
 
       result.fold(
         { e =>
           logger.warn("Could not subscribe", e)
           errorHandler.errorResult()
-        }, { subscriptionResponse =>
-          logger.info(s"Successfully subscribed with cgt id ${subscriptionResponse.cgtReferenceNumber}")
-          SeeOther(routes.SubscriptionController.subscribed().url)
+        }, {
+          case SubscriptionResponse.SubscriptionSuccessful(cgtReferenceNumber) =>
```
Comment (co-category) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/SubscriptionController.scala]: Worth importing the values of `SubscriptionResponse` unless you think the semantics are clearer this way

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/SubscriptionController.scala]: I had no real reason of diong it this way - I've changed it now

**Pull request merged**